packages/app-mobile/root.tsx

```diff
@@ -207,7 +207,10 @@ const generalMiddleware = (store: any) => (next: any) => async (action: any) =>
 		setTimeLocale(Setting.value('locale'));
 	}
 
-	if ((action.type === 'SETTING_UPDATE_ONE' && (action.key.indexOf('encryption.') === 0)) || (action.type === 'SETTING_UPDATE_ALL')) {
+	// Like the desktop and CLI apps, we run this whenever the sync target properties change.
+	// Previously, this only ran when encryption was enabled/disabled. However, after fetching
+	// a new key, this needs to run.
+	if ((action.type === 'SETTING_UPDATE_ONE' && action.key === 'syncInfoCache') || (action.type === 'SETTING_UPDATE_ALL')) {
```

Comment:
**Note**: Similar logic in `BaseApplication.ts` had a similar change applied [in 2021](https://github.com/laurent22/joplin/blame/5062cd161680557ad697a2be1669deaebe5c65b4/packages/lib/BaseApplication.ts#L387).